### DISCUSSION TITLE
Twinery Integration

### DISCUSSION THREAD

**Discussion by atebyagrue** [evennia/evennia#3619] [Community Contribs & Snippets]
I'm considering taking a look at creating a mod as an integration to [twinery](https://twinery.org/). Twine is an awesome product I've used in past game development projects to build out dialog trees & track conversational branching. The current webapp (ver 3) now allows for an export into a .twee file. I imagine this file could be parsed & then a Evennia evmenu file automatically be generated; based on the .twee file's contents. Has anyone already gone down this path? Is this something other content creators would use?

**Comment by InspectorCaracal** [comment]
I love twine! I think this could be an interesting project.

The biggest challenge I see for it is that Twine and Evennia are built on extremely different architectures and using very different goals and design principles, but I haven't tried out v3 yet so these .twee files are new to me! No idea what they look like inside yet, it might be easy. xD

A good place to start on the Evennia side though is to check out how it parses the existing plaintext "menu template" style of generating a menu; I think that'll be the most similar process to what you'd aiming for.

**Comment by Griatch** [comment]
Twine is cool, I don't have much to add beyond what InspectorCaracal already wrote; EvMenu already allows for being built from a template file, so use that for inspiration :+1:
